engines/sherlock/animation.cpp

```diff
@@ -0,0 +1,203 @@
+static const char *const PROLOGUE_NAMES[6] = {
+	"subway1", "subway2", "finale2", "suicid", "coff3", "coff4"
+};
+static const int PROLOGUE_FRAMES[6][9] = {
+	{ 4, 26, 54, 72, 92, 134, FRAMES_END },
+	{ 2, 80, 95, 117, 166, FRAMES_END },
+	{ 1, FRAMES_END },
+	{ 42, FRAMES_END },
+	{ FRAMES_END },
+	{ FRAMES_END }
+};
+
+// Title animations file list
+static const char *const TITLE_NAMES[7] = {
+	"27pro1", "14note", "coff1", "coff2", "coff3", "coff4", "14kick"
+};
+
+static const int TITLE_FRAMES[7][9] = {
+	{ 29, 131, FRAMES_END },
+	{ 55, 80, 95, 117, 166, FRAMES_END },
+	{ 15, FRAMES_END },
+	{ 4, 37, 92, FRAMES_END },
+	{ 2, 43, FRAMES_END },
+	{ 2, FRAMES_END },
+	{ 10, 50, FRAMES_END }
+};
+
+static const int NO_FRAMES = FRAMES_END;
+
+Animation::Animation(SherlockEngine *vm): _vm(vm) {
+}
+
+/**
+ * Play a full-screen animation
+ */
+bool Animation::play(const Common::String &filename, int minDelay, int fade,
+		bool setPalette, int speed) {
+	Events &events = *_vm->_events;
+	Screen &screen = *_vm->_screen;
+	Sound &sound = *_vm->_sound;
+	int soundNumber = 0;
+	sound._playingEpilogue = true;
+
+	// Check for any any sound frames for the given animation
+	const int *soundFrames = checkForSoundFrames(filename);
+
+	// Strip any extension off of the passed filename and add .vdx suffix
+	Common::String baseName = filename;
+	const char *p = strchr(baseName.c_str(), '.');
+	if (p)
+		baseName = Common::String(filename.c_str(), MIN(p - 1, baseName.c_str() + 7));
+
+	Common::String vdxName = baseName + ".vdx";
+
+	// Load the animation
+	Common::SeekableReadStream *stream;
+	if (!_vm->_titleOverride.empty())
+		stream = _vm->_res->load(vdxName, _vm->_titleOverride);
+	else if (_vm->_useEpilogue2)
+		stream = _vm->_res->load(vdxName, "epilog2.lib");
+	else
+		stream = _vm->_res->load(vdxName, "epilogue.lib");
+
+	// Load initial image
+	Common::String vdaName = baseName + ".vda";
+	ImageFile images(vdaName, true, true);
+
+	events.wait(minDelay);
+	if (fade != 0 && fade != 255)
+		screen.fadeToBlack();
+
+	if (setPalette) {
+		if (fade != 255)
+			screen.setPalette(images._palette);
+	}
+
+	int frameNumber = 0;
+	Common::Point pt;
+	bool skipped = false;
+	while (!_vm->shouldQuit()) {
+		// Get the next sprite to display
+		int imageFrame = stream->readSint16LE();
+
+		if (imageFrame == -2) {
+			// End of animation reached
+			break;
+		} else if (imageFrame != -1) {
+			// Read position from either animation stream or the sprite frame itself
+			if (imageFrame < 0) {
+				imageFrame += 32768;
+				pt.x = stream->readUint16LE();
+				pt.y = stream->readUint16LE();
+			} else {
+				pt = images[imageFrame]._offset;
+			}
+
+			// Draw the sprite. Note that we explicitly use the raw frame below, rather than the ImageFrame,
+			// since we don't want the offsets in the image file to be used, just the explicit position we specify
+			screen.transBlitFrom(images[imageFrame]._frame, pt);
+		} else {
+			// No sprite to show for this animation frame
```

Comment:
This comment looks wrong. You're only incrementing frameNumber inside this case (imageFrame == -1), so I think it's more likely that this is a 'frame done' marker.